### PR TITLE
Feature/delete downloads

### PR DESCRIPTION
**Pull request by JeltevanBoheemen**
This partially implements #976, only the download part. Even with an expiration period, it's nice to offer a direct way. Creating a PR now because deleting accounts is slightly more involved, and may take some time to get around to for me.

Tried really hard to fit the `delete all` button underneath the `Delete` header, but couldn't get PrimeNG to comply.

### CODE REVIEW

File: frontend/src/app/history/download-history/download-history.component.html [hunk 1]
```diff
@@ -7,6 +7,12 @@ <h1 class="title">Downloads</h1>
                 Downloads
             </div>
         </ng-template>
+        <ng-template *ngIf="downloads.length" pTemplate="summary">
+            <button class="button is-primary" type="reset" (click)="deleteAllDownloads()">
+                <span class="icon"><fa-icon [icon]="faTrash"></fa-icon></span>
+                <span>Delete all downloads</span>
+            </button>
+        </ng-template>
```
Comment (lukavdplas) [inline frontend/src/app/history/download-history/download-history.component.html:15]: I agree this is a good place for a delete button, but it's inconsistent that downloads are deleted here, while queries are deleted from the settings menu. The pages are almost identical, so they should handle this the same way. Perhaps you could move the "delete search history" button?

File: frontend/src/app/history/download-history/download-history.component.html [hunk 2]
```diff
@@ -48,6 +55,16 @@ <h1 class="title">Downloads</h1>
...
+                <td>
+                    <a *ngIf="download.status === 'done' || download.status ==='error'" role="button" tabindex="0"
+                        (click)="deleteDownload(download)"
+                        (keydown.enter)="deleteDownload(download)">
+                        <span class="icon">
+                            <fa-icon [icon]="faTrash"></fa-icon>
+                        </span>
+                        <span>delete download</span>
+                    </a>
+                </td>
```
Comment (lukavdplas) [inline frontend/src/app/history/download-history/download-history.component.html:67]: Nice.

Addition my earlier comment: if you choose to make the search history menu more like this page, don't bother adding this action. The search history includes a unique filter, which means deleting one query would usually cause another identical one to appear. A decent delete button per query would take more time.

**Review by lukavdplas — CHANGES_REQUESTED** [review]
Nice, very neat! :)

I have one comment re. the interface in the downloads and search history page.

**Comment by JeltevanBoheemen** [comment]
Looking at this later!

**Pull request closed without merging**